packages/Uploader/src/Uploader.js

```diff
@@ -0,0 +1,219 @@
+  /**
+    When true the user will be able to drop files at any part of the document.body triggering
+    the upload listeners. On false will only received files if they are drop exactly on the FileInput area.
+  */
+  isBodyDroppableArea: PropTypes.bool,
+  /**
+    Size in Mebibytes which is use for comparing each file that will be upload
+    you can make use of Uploader.convertUnitsToMebibytes() for easily convert units to Mebibyes
+    ex.  Uploader.convertUnitsToMebibytes(4) => 4194304 (close to 4MB);
+  */
+  maximumFileSize: PropTypes.number,
+  /**
+    This callback fires every time a file has been processed
+  */
+  onChange: PropTypes.func,
+  /**
+    Will be fire once all files has been processed with the files as parameter, finishing doesn't mean that all
+    files has been correctly uploaded, only means that they were processed, to find about the state of each file
+    iterate over the files list.
+  */
+  onFinished: PropTypes.func,
+  /**
+    you can pass an array of header objects as need it ex.
+    <Uploader headers={[{ "API-Key": "your-api-key" }, { Accept: "application/json" }, { "X-CSRF-Token": "your-token"} ]} ...>
+      {({...}) => <YourUI />}
+    </Uploader>
+  */
+  headers: PropTypes.arrayOf(PropTypes.object),
+};
+
+const defaultProps = {
+  a11yText: null,
+  acceptableFileTypes: ["*/*"],
+  allowMultipleFile: true,
+  defaultIsDisable: false,
+  hasAutoupload: true,
+  headers: [],
+  isBodyDroppableArea: true,
+  maximumFileSize: oneMebibyte * 10, // 1048576bytes * 10 = 10,485,760 Mebibytes
+  onChange: () => {},
+  onFinished: () => {},
+};
+
+function getDocumentBody() {
+  return document.body;
+}
+
+function getContainer(refContainer) {
+  return function findContainer() {
+    return refContainer.current;
+  };
+}
+
+function UploaderComponent(props, ref) {
+  const {
+    a11yText,
+    acceptableFileTypes,
+    allowMultipleFile,
+    hasAutoupload,
+    maximumFileSize,
+    onChange,
+    onFinished,
+    isBodyDroppableArea,
+    endpoint,
+    children,
+    defaultIsDisable,
+    headers,
+  } = props;
+
+  const refInput = React.useRef();
+  const refContainer = React.useRef(null);
+  const [refId] = React.useState(uuidv4());
+  const i18n = useI18n();
+  const label = a11yText || i18n.t("uploader.label");
+
+  React.useImperativeHandle(ref, () => ({
+    focus: () => {
+      refInput.current.focus();
+    },
+  }));
+
+  const { files, hasFinished, isDisabled, removeFile, cancelFile, setFiles, upload } = useProcessFiles({
+    defaultIsDisable,
+    endpoint,
+    hasAutoupload,
+    headers,
+    onChange,
+    onFinished,
+  });
+
+  function handleChange(event) {
+    if (isDisabled) return;
+
+    const files = getFiles({ event, maximumFileSize, acceptableFileTypes, endpoint });
+    setFiles(() => {
+      if (refInput.current) {
+        refInput.current.value = "";
+      }
+      return allowMultipleFile ? files : [files[0]]; // in case only allow one file per upload
```

Comment:
Should there be an error displayed in the case of `allowMultipleFile`, and `file.length > 1`?

Reply:
I'm leaning to think we don't need it 🤔.  That error wouldn't occur if the user select files via the file input, because the input automatically will only let her pick one file. 

This is relevant when a user drop multiple files, in that case, I'm just taking the first one, is a little edge casie, not sure if we want to show/display an error.